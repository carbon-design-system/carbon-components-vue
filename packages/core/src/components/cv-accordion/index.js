import CvAccordion from './cv-accordion';
import CvAccordionItem from './cv-accordion-item';
import CvAccordionSkeleton from './cv-accordion-skeleton';

export { CvAccordion, CvAccordionItem, CvAccordionSkeleton };
