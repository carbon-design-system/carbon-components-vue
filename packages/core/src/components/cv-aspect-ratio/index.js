import CvAspectRatio from './cv-aspect-ratio';
export { CvAspectRatio };
