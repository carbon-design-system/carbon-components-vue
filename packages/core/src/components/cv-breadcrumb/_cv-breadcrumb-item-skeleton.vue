<template>
  <div :class="`${carbonPrefix}--breadcrumb-item`">
    <a :class="`${carbonPrefix}--link`">&nbsp;</a>
  </div>
</template>

<script>
import { carbonPrefixMixin } from '../../mixins';
export default {
  name: 'CvBreadcrumbItemSkeleton',
  mixins: [carbonPrefixMixin],
};
</script>
