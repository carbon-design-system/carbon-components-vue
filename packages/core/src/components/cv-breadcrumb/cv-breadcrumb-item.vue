<template>
  <div
    :class="[
      `cv-breadcrumb-item ${carbonPrefix}--breadcrumb-item`,
      { [`${carbonPrefix}--breadcrumb-item--current`]: activePage },
    ]"
    :aria-current="ariaCurrent"
  >
    <slot></slot>
  </div>
</template>

<script>
import { carbonPrefixMixin } from '../../mixins';
export default {
  name: 'CvBreadcrumbItem',
  mixins: [carbonPrefixMixin],
  props: {
    active: Boolean,
    ariaCurrent: String,
  },
  computed: {
    activePage() {
      return this.active && this.ariaCurrent !== 'page';
    },
  },
};
</script>
