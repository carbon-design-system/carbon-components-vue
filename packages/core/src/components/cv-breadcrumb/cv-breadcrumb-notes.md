# CvBreadcrumb

A Vue implementation of a Carbon Components breadcrumb

http://www.carbondesignsystem.com/components/breadcrumb/code

## Usage

```html
<cv-breadcrumb no-trailing-slash>
  <cv-breadcrumb-item>Abc</cv-breadcrumb-item>
  <cv-breadcrumb-item>Def</cv-breadcrumb-item>
  <cv-breadcrumb-item>ghi</cv-breadcrumb-item>
</cv-breadcrumb>
```

Often used with breadcrumbs or other controls

```html
<cv-breadcrumb no-trailing-slash>
  <cv-breadcrumb-item>
    <cv-breadcrumb href="#outer">outer-breadcrumb</cv-breadcrumb>
  </cv-breadcrumb-item>
  <cv-breadcrumb-item>
    <cv-breadcrumb href="#outer">parent-breadcrumb</cv-breadcrumb>
  </cv-breadcrumb-item>
  <cv-breadcrumb-item>
    <cv-text value="name of thing"></cv-text>
  </cv-breadcrumb-item>
</cv-breadcrumb>
```

## Attributes

aria-label: changes the breadcrumb aria label
no-trailing-slash: if true no trailing slash follows the last breadcrumb item

## Item attributes

active: is the active or current location
