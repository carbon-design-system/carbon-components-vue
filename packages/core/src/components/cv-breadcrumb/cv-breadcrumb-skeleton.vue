<template>
  <nav :class="`${carbonPrefix}--breadcrumb ${carbonPrefix}--skeleton`" :aria-label="ariaLabel">
    <cv-breadcrumb-item-skeleton></cv-breadcrumb-item-skeleton>
    <cv-breadcrumb-item-skeleton></cv-breadcrumb-item-skeleton>
    <cv-breadcrumb-item-skeleton></cv-breadcrumb-item-skeleton>
  </nav>
</template>

<script>
import CvBreadcrumbItemSkeleton from './_cv-breadcrumb-item-skeleton';
import { carbonPrefixMixin } from '../../mixins';

export default {
  name: 'CvBreadcrumbSkeleton',
  mixins: [carbonPrefixMixin],
  components: {
    CvBreadcrumbItemSkeleton,
  },
  props: {
    ariaLabel: { type: String, default: 'breadcrumb' },
  },
};
</script>
