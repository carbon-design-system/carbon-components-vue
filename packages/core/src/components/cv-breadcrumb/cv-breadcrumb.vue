<template>
  <nav
    class="cv-breadcrumb"
    :class="[`${carbonPrefix}--breadcrumb`, { [`${carbonPrefix}--breadcrumb--no-trailing-slash`]: noTrailingSlash }]"
    :aria-label="ariaLabel"
  >
    <slot></slot>
  </nav>
</template>

<script>
import { carbonPrefixMixin } from '../../mixins';
export default {
  name: 'CvBreadcrumb',
  mixins: [carbonPrefixMixin],
  props: {
    ariaLabel: { type: String, default: 'breadcrumb' },
    noTrailingSlash: Boolean,
  },
};
</script>
