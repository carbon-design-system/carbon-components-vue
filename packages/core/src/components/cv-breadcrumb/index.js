import CvBreadcrumb from './cv-breadcrumb';
import CvBreadcrumbItem from './cv-breadcrumb-item';
import CvBreadcrumbSkeleton from './cv-breadcrumb-skeleton';

export { CvBreadcrumb, CvBreadcrumbItem, CvBreadcrumbSkeleton };
