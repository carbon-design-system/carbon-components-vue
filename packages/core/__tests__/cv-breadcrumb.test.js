// import { shallowMount as shallow } from '@vue/test-utils';
import { testComponent, awaitNextTick } from './_helpers';
const { shallowMount: shallow } = awaitNextTick;
import { CvBreadcrumb, CvBreadcrumbSkeleton, CvBreadcrumbItem } from '@/components/cv-breadcrumb';
// import { settings as carbonSettings } from 'carbon-components';

// const carbonPrefix = carbonSettings.prefix;

describe('CvBreadcrumb', () => {
  // ***************
  // PROP CHECKS
  // ***************
  testComponent.propsAreType(CvBreadcrumb, ['ariaLabel'], String);
  testComponent.propsAreType(CvBreadcrumb, ['noTrailingSlash'], Boolean);

  testComponent.propsHaveDefault(CvBreadcrumb, ['ariaLabel']);
  testComponent.propsHaveDefault(CvBreadcrumbSkeleton, ['ariaLabel']);

  // ***************
  // SNAPSHOT TESTS
  // ***************
  it('matches breadcrumb with trailing slash', async () => {
    const propsData = { noTrailingSlash: false };
    const wrapper = await shallow(CvBreadcrumb, { propsData });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('matches breadcrumb without trailing slash', async () => {
    const propsData = { noTrailingSlash: true };
    const wrapper = await shallow(CvBreadcrumb, { propsData });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('matches breadcrumb with slotted content', async () => {
    const wrapper = await shallow(CvBreadcrumb, {
      slots: {
        default: CvBreadcrumbItem,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  // ***************
  // FUNCTIONAL TESTS
  // ***************
});

describe('CvBreadcrumbItem', () => {
  // ***************
  // PROP CHECKS
  // ***************

  // ***************
  // SNAPSHOT TESTS
  // ***************
  it('matches breadcrumb item with slotted content', async () => {
    const wrapper = await shallow(CvBreadcrumbItem);

    expect(wrapper.html()).toMatchSnapshot();
  });
  // ***************
  // FUNCTIONAL TESTS
  // ***************
});

describe('CvBreadcrumbSkeleton', () => {
  // ***************
  // PROP CHECKS
  // ***************

  // ***************
  // SNAPSHOT TESTS
  // ***************
  it('matches breadcrumb item with slotted content', async () => {
    const wrapper = await shallow(CvBreadcrumbSkeleton);

    expect(wrapper.html()).toMatchSnapshot();
  });
  // ***************
  // FUNCTIONAL TESTS
  // ***************
});
