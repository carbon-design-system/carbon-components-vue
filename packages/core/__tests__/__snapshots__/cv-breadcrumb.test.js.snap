// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CvBreadcrumb matches breadcrumb with slotted content 1`] = `
<nav aria-label="breadcrumb" class="cv-breadcrumb bx--breadcrumb">
  <div class="cv-breadcrumb-item bx--breadcrumb-item"></div>
</nav>
`;

exports[`CvBreadcrumb matches breadcrumb with trailing slash 1`] = `<nav aria-label="breadcrumb" class="cv-breadcrumb bx--breadcrumb"></nav>`;

exports[`CvBreadcrumb matches breadcrumb without trailing slash 1`] = `<nav aria-label="breadcrumb" class="cv-breadcrumb bx--breadcrumb bx--breadcrumb--no-trailing-slash"></nav>`;

exports[`CvBreadcrumbItem matches breadcrumb item with slotted content 1`] = `<div class="cv-breadcrumb-item bx--breadcrumb-item"></div>`;

exports[`CvBreadcrumbSkeleton matches breadcrumb item with slotted content 1`] = `
<nav aria-label="breadcrumb" class="bx--breadcrumb bx--skeleton">
  <cv-breadcrumb-item-skeleton-stub></cv-breadcrumb-item-skeleton-stub>
  <cv-breadcrumb-item-skeleton-stub></cv-breadcrumb-item-skeleton-stub>
  <cv-breadcrumb-item-skeleton-stub></cv-breadcrumb-item-skeleton-stub>
</nav>
`;
